broadcaster/sequencenumbercatchupbuffer.go

```diff
@@ -94,7 +94,7 @@ func (b *SequenceNumberCatchupBuffer) deleteConfirmed(confirmedSequenceNumber ar
 	confirmedIndex := uint64(confirmedSequenceNumber - firstSequenceNumber)
 
 	if confirmedIndex >= uint64(len(b.messages)) {
-		log.Error("ConfirmedSequenceNumber is past the end of stored messages", "confirmedSequenceNumber", confirmedSequenceNumber, "firstSequenceNumber", firstSequenceNumber, "cacheLength", len(b.messages))
+		log.Warn("ConfirmedSequenceNumber is past the end of stored messages", "confirmedSequenceNumber", confirmedSequenceNumber, "firstSequenceNumber", firstSequenceNumber, "cacheLength", len(b.messages))
```

Comment:
I'm not sure exactly how this gets hit, but could we maybe only log if `confirmedIndex > uint64(len(b.messages))` but still clear the messages if they're equal? I don't think this is user input triggered though, I think it's caused by the upstream feed confirming something before it's sent.

Reply:
Yes, not user input triggered.  `index == len()` is still past the end of the array, so still invalid.  Only occurs when standby sequencer is catching up and listening to primary feed.

Reply:
I'm going to mark this PR as a draft as we're currently waiting to see if the feed reorg PR https://github.com/OffchainLabs/nitro/pull/1230 fixes the issue, and if not, we need to figure out the root cause of this